### DISCUSSION TITLE
Toggle Group Summaries with One Button

### DISCUSSION THREAD

**Discussion by gfo2007** [serenity-is/Serenity#6711] [Help]
Hi,
I have followed the demo to implement the Group Summaries. However, instead of using one button to group and another to ungroup, I would like to use only one button to toggle between grouping and ungrouping. Below is the image and code. Kindly help with what needs to change.

```
buttons.push({
                title: "Group by Material Type",
                cssClass: "expand-all-button",
                //separator: true,
                onClick: () => this.view.setGrouping
                    ([{
                        getter: fld.MaterialTypeMaterialTypeName,
                        // had to add this line bellow
                        //aggregators: this.aggregators
                    }])
            });
```

![Group Summary Toggle](https://user-images.githubusercontent.com/14317420/222952405-89ed4cdf-0bfc-4932-b880-524eb9abe581.PNG)

**Comment by wezmag** [comment]
I think you can use `this.view.getGrouping()` to check if grouping has been set to the grid.
